src/cargo/core/shell.rs

```diff
@@ -1,17 +1,45 @@
 use std::fmt;
 use std::io::prelude::*;
 use std::io;
+use std::error::Error;
 
 use term::Attr;
 use term::color::{Color, BLACK, RED, GREEN, YELLOW};
 use term::{Terminal, TerminfoTerminal, color};
 
 use self::AdequateTerminal::{NoColor, Colored};
+use self::Verbosity::{Verbose, Normal, Quiet};
+
+use util::{CargoResult, CargoError};
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum Verbosity {
+    Verbose,
+    Normal,
+    Quiet
+}
+
+#[derive(Debug)]
+pub struct VerbosityError;
```

Comment:
One-off errors like this can typically be created with `human("cannot set both --verbose and --quiet")`